packages/template-retail-react-app/app/hooks/use-wish-list.js

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (c) 2022, Salesforce, Inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import {useProductLists} from './use-product-lists'
+import {useProducts} from 'commerce-sdk-react-preview'
+
+export const useWishList = ({listId = '', shouldFetchProductDetail = false} = {}) => {
+    const {
+        productLists,
+        error: productListsError,
+        isLoading: isProductListsLoading
+    } = useProductLists()
+    const wishLists = productLists?.data?.filter((list) => list.type === 'wish_list')
+    // current use wish list
+    const wishlist = wishLists?.data?.find((list) => list.id === listId) || productLists?.data?.[0]
```

Comment:
Minor: if we need to add a comment here, IMO it's better to update the variable name like this instead:

```suggestion
    const currentWishlist = wishLists?.data?.find((list) => list.id === listId) || productLists?.data?.[0]
```